core/module.go

```diff
@@ -2,51 +2,64 @@ package core
 
 import (
 	"context"
-	"encoding/json"
 	"fmt"
 	"log/slog"
-	"path/filepath"
+	"sort"
 	"strings"
 	"time"
 
-	"github.com/dagger/dagger/core/modules"
 	"github.com/dagger/dagger/dagql"
 	"github.com/dagger/dagger/dagql/idproto"
 	"github.com/moby/buildkit/solver/pb"
+	"github.com/pkg/errors"
 	"github.com/vektah/gqlparser/v2/ast"
+	"golang.org/x/sync/errgroup"
 )
 
 type Module struct {
 	Query *Query
 
+	// The source of the module
+	Source dagql.Instance[*ModuleSource] `field:"true" name:"source" doc:"The source for the module."`
```

Comment:
Super nice having this available, making this the canonical ID for deduping makes sense to me